src/Environment.php

```diff
@@ -58,9 +61,12 @@ public function installed(): bool
     public function identifyHostname()
     {
         $this->app->singleton(CurrentHostname::class, function () {
-            $hostname = $this->dispatch(new HostnameIdentification);
 
-            return $hostname;
+            if ($this->runningInConsole() || !$this->installed()) {
```

Comment:
Only issue with doing it this way vs how I suggested in #366, is a very minor point of efficiency.  This method will make 1-2 queries (1 to check for table existence, and (possibly) 1 to identify the hostname).  The method I suggested will ALWAYS require a single query, and can just catch the "table not found" database exception and return null.

It's probably a moot point given how small the queries are.

Reply:
Good point.